packages/flutter/test/widgets/navigator_test.dart

```diff
@@ -5099,6 +5099,140 @@ void main() {
         variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.android }),
         skip: isBrowser, // [intended] only non-web Android supports predictive back.
       );
+
+      testWidgets('popping a page with canPop true still calls onPopInvoked', (WidgetTester tester) async {
+        final List<_PageWithYesPop> pages = <_PageWithYesPop>[_PageWithYesPop.home];
+        bool canPop() => pages.length <= 1;
+        bool onPopInvokedWasCalled = false;
```

Comment:
Maybe turn this into a count, so we can be sure it is only invoked once and not multiple times via different code paths?

Reply:
That's a much better idea, thanks.